src/main/java/core/basesyntax/service/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private StorageDao storageDao;
+
+    public PurchaseOperationHandler() {
+        storageDao = new StorageDaoImpl();
```

Comment:
Consider using dependency injection to provide the `StorageDao` instance. This approach can improve testability and flexibility by allowing different implementations of `StorageDao` to be used without modifying the handler.